org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/xsd/participants/XSDRenameParticipant.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4xml.extensions.xsd.participants;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4xml.commons.BadLocationException;
+import org.eclipse.lsp4xml.dom.DOMAttr;
+import org.eclipse.lsp4xml.dom.DOMDocument;
+import org.eclipse.lsp4xml.dom.DOMElement;
+import org.eclipse.lsp4xml.dom.DOMNode;
+import org.eclipse.lsp4xml.extensions.xsd.utils.XSDUtils;
+import org.eclipse.lsp4xml.services.extensions.IRenameParticipant;
+import org.eclipse.lsp4xml.services.extensions.IRenameRequest;
+import org.eclipse.lsp4xml.utils.DOMUtils;
+import org.eclipse.lsp4xml.utils.XMLPositionUtility;
+
+/**
+ * XSD rename
+ * 
+ */
+public class XSDRenameParticipant implements IRenameParticipant {
+
+	@Override
+	public void doRename(IRenameRequest request, List<TextEdit> locations) {
+		DOMDocument xmlDocument = request.getDocument();
+
+		if (!DOMUtils.isXSD(xmlDocument)) {
+			return;
+		}
+		for (TextEdit textEdit: getRenameTextEdits(request)) {
+			locations.add(textEdit);
+		}
+
+	}
+
+	private List<TextEdit> getRenameTextEdits(IRenameRequest request) {
+		DOMDocument document = request.getDocument();
+		DOMElement element = request.getElement();
+		Position position = request.getPosition();
+		String newText = request.getNewText();
+
+		if (XSDUtils.isXSComplexType(element)) {
```

Comment:
replace with if (XSDUtils.isXSComplexType(element) || XSDUtils.isXSSimpleType(element)) {

xs:simpleType should be renamed too.  I suggest you try that with https://github.com/angelozerr/lsp4xml/blob/master/org.eclipse.lsp4xml/src/test/resources/xsd/invoice-ns.xsd#L45